source/rules/no-unsafe-switchmap.ts

```diff
@@ -0,0 +1,165 @@
+    const defaultDisallow = [
+      "add",
+      "create",
+      "delete",
+      "post",
+      "put",
+      "remove",
+      "set",
+      "update"
+    ];
+
+    function createRegExp(value: any): RegExp | null {
+      if (!value || !value.length) {
+        return null;
+      }
+      const flags = "i";
+      if (typeof value === "string") {
+        return new RegExp(value, flags);
+      }
+      const words = value as string[];
+      const joined = words.map(word => `(\\b|_)${word}(\\b|_)`).join("|");
+      return new RegExp(`(${joined})`, flags);
+    }
+
+    let allowRegExp: RegExp | null;
+    let disallowRegExp: RegExp | null;
+    let observableRegExp: RegExp;
+
+    const [config] = context.options;
+    if (config && (config.allow || config.disallow)) {
+      allowRegExp = createRegExp(config.allow);
+      disallowRegExp = createRegExp(config.disallow);
+      observableRegExp = new RegExp(config.observable || defaultObservable);
+    } else {
+      allowRegExp = null;
+      disallowRegExp = createRegExp(defaultDisallow);
+      observableRegExp = new RegExp(defaultObservable);
+    }
+
+    const { couldBeObservable, isReferenceType } = typecheck(context);
+
+    function shouldDisallow(args: es.Node[]): boolean {
+      const names = args
+        .map(arg => {
+          if (isLiteral(arg) && typeof arg.value === "string") {
+            return arg.value;
+          }
+          if (isIdentifier(arg)) {
+            return arg.name;
+          }
+
+          return "";
+        })
+        // tslint:disable-next-line: no-unnecessary-callback-wrapper
+        .map(name => decamelize(name));
+
+      if (allowRegExp) {
+        return !names.every(name => allowRegExp.test(name));
+      }
+      if (disallowRegExp) {
+        return names.some(name => disallowRegExp.test(name));
+      }
+
+      return false;
+    }
+
+    function report(node: es.CallExpression) {
+      if (
+        !node.arguments ||
+        !isReferenceType(node) ||
+        !couldBeObservable(node)
+      ) {
+        return;
+      }
+
+      const hasInvalidOfType = node.arguments.some(arg => {
+        if (
+          isCallExpression(arg) &&
+          isIdentifier(arg.callee) &&
+          arg.callee.name === "ofType"
+        ) {
+          return shouldDisallow(arg.arguments);
+        }
+      });
+
+      if (hasInvalidOfType) {
+        const switchMapNodes = esquery(
+          node,
+          "[arguments] > CallExpression > Identifier[name='switchMap']"
+        ) as es.Identifier[];
+        switchMapNodes.forEach(switchMapNode => {
+          context.report({
+            messageId: "forbidden",
+            node: switchMapNode
+          });
+        });
+      }
+    }
+
+    return {
+      [`CallExpression[callee.property.name='pipe'][callee.object.name=${observableRegExp}]`]: report,
+      [`CallExpression[callee.property.name='pipe'][callee.object.property.name=${observableRegExp}]`]: report
```

Comment:
Won't the regular expressions returned by `createRegExp` include the `i` flag? How can they be converted to strings here if `esquery` does not support regular expressions with flags?

Reply:
The regex expressions created with `createRegExp` are `allowRegExp` and `disallowRegExp`. We don't use these expressions in a selector but we're using the `test` function, that's why these aren't given any problems.

https://github.com/cartant/eslint-plugin-rxjs/pull/35/files#diff-34689f0387e32bc475a66f4b7007e70dR107